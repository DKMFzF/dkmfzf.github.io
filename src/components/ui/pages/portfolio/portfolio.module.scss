@use '~styles/variables' as *;

.container {
	position: relative;
	width: 100vw;
	height: 85vh;
	overflow: hidden;
	background-color: #f5f5f5;
}

.mobileMessage {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100vw;
	height: 100vh;
	padding: 20px;
	color: $bg-primary-color;
	font-weight: bold;
	font-size: 24px;
	font-family: $primary-font;
	text-align: center;
	opacity: $opacity;
}

.heading {
	margin: 20px;
	font-weight: bold;
	font-size: 24px;
}
