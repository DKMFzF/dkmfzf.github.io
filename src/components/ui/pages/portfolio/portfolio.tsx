import React from 'react';
import { useDragZoom } from '@hooks';
import styles from './portfolio.module.scss';
import { CardsProject } from '@components';
import { portfolioCard } from '@utils-constants';
import { useCards } from '@hooks';

export const PortfolioPageUI: React.FC = () => {
	const { isDragging, position, scale, handleMouseDown, containerRef } =
		useDragZoom();

	const { cards: positionedCards } = useCards();

	const cardsWithPosition = portfolioCard.map((originalCard, index) => {
		const positionData =
			positionedCards[Math.min(index, positionedCards.length - 1)];

		return {
			...originalCard,
			id: index,
			x: positionData.x,
			y: positionData.y,
			rotation: positionData.rotation,
			width: positionData.width,
			height: positionData.height
		};
	});

	return (
		<main
			ref={containerRef}
			className={styles.container}
			style={{ cursor: isDragging ? 'grabbing' : 'grab' }}
			onMouseDown={handleMouseDown}
		>
			<CardsProject
				cards={cardsWithPosition}
				transformStyle={`translate(${position.x}px, ${position.y}px) scale(${scale})`}
			/>
		</main>
	);
};
