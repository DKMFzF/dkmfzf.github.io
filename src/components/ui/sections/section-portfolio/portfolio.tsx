import { FC } from 'react';
import { SectionProfileUIProps } from './type';

export const SectionPortfolioUI: FC<SectionProfileUIProps> = () => (
	<section id='SectionPortfolioUI'>Привет мир!</section>
);
