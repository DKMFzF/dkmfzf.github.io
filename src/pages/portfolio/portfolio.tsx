import { FC } from 'react';
import { PortfolioPageUI } from 'src/components/ui';

export const PortfolioPage: FC = () => <PortfolioPageUI />;
