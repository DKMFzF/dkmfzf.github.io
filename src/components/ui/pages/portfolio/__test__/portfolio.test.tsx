import { render } from '@testing-library/react';
import { HelmetProvider } from 'react-helmet-async';
import { expect } from '@jest/globals';
import { testRenderComponent } from '@utils-testing';
import { PortfolioPageUI } from '../portfolio';

jest.mock('@hooks', () => ({
	__esModule: true,
	useDragZoom: jest.fn(() => ({
		isDragging: false,
		position: { x: 0, y: 0 },
		scale: 1,
		handleMouseDown: jest.fn(),
		containerRef: { current: null }
	})),
	useCards: jest.fn(() => ({
		cards: [
			{ x: 0, y: 0, rotation: 0, width: 100, height: 100 },
			{ x: 100, y: 100, rotation: 10, width: 100, height: 100 }
		]
	}))
}));

jest.mock('@components', () => ({
	__esModule: true,
	CardsProject: jest.fn(() => <div>CardsProject Mock</div>)
}));

describe('[PortfolioPageUI]', () => {
	beforeEach(() => {
		Object.defineProperty(window, 'innerWidth', {
			writable: true,
			configurable: true,
			value: 1920
		});
	});

	it('should render desktop view correctly', () => {
		testRenderComponent(
			<HelmetProvider>
				<PortfolioPageUI />
			</HelmetProvider>
		);
	});

	it('should render mobile view correctly', () => {
		Object.defineProperty(window, 'innerWidth', {
			writable: true,
			configurable: true,
			value: 800
		});

		testRenderComponent(
			<HelmetProvider>
				<PortfolioPageUI />
			</HelmetProvider>
		);
	});
});
