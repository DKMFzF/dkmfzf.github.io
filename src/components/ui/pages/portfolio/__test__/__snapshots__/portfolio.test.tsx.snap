// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PortfolioPageUI] should render desktop view correctly 1`] = `
<DocumentFragment>
  <main
    class="container"
    style="cursor: grab;"
  >
    <div>
      CardsProject Mock
    </div>
  </main>
</DocumentFragment>
`;

exports[`[PortfolioPageUI] should render mobile view correctly 1`] = `
<DocumentFragment>
  <main
    class="mobile-message"
    data-cy="content"
    style=""
  >
    Please open this page from your PC
  </main>
</DocumentFragment>
`;
